### PR TITLE
fix(mqtt-ad): expose Scene Activation CC

### LINKED ISSUE #3570: Scene Activation CC (43) support with MQTT Discovery
### Version

Build/Run method:

[ X] Docker
[ ] Git Clone + npm install + npm run build + npm start
[ ] Pkg

Z-Wave JS UI version: 9.8.2
Z-Wave JS version: 12.4.3

### Describe the bug

I have a ZME_RC2  remote controller. It was working well until I upgrade to the last version (I don't know which version I was using before, one from the end of 2023), 
After troubleshooting, I ve found that the change states are not published to "right" mqtt topic.
In home assistant section, i see hass device json with command topic like : 
"command_topic": "zwave/27/112/0/10/set",

But when I check with mqtt explorer, when the device change, the value is set in "zwave/27/43/0/"

Here is debug information when I press a key on remote controller
```
2024-01-31 20:56:38.707 DEBUG MQTT: Publishing to zwave/27/lastActive: { time: 1706734598707, value: 1706734598706 } with options { qos: 1, retain: false }
2024-01-31 20:56:38.722 DEBUG MQTT: Publishing to zwave/27/43/0/sceneId: {
  time: 1706734598721,
  value: 72,
  nodeName: 'Telecommande',
  nodeLocation: ''
} with options { qos: 1, retain: false }
2024-01-31 20:56:38.724 INFO Z-WAVE: [Node 027] Value notification: 43-0-sceneId 72
2024-01-31 20:56:38.729 DEBUG MQTT: Publishing to zwave/27/43/0/dimmingDuration: {
  time: 1706734598726,
  value: Duration { unit: 'default', _value: 0, [value]: [Getter/Setter] },
  nodeName: 'Telecommande',
  nodeLocation: ''
} with options { qos: 1, retain: false }
2024-01-31 20:56:38.731 INFO Z-WAVE: [Node 027] Value updated: 43-0-dimmingDuration default => default
2024-01-31 20:56:38.959 DEBUG MQTT: Publishing to zwave/27/lastActive: { time: 1706734598959, value: 1706734598706 } with options { qos: 1, retain: false }
2024-01-31 20:56:39.724 DEBUG MQTT: Publishing to zwave/27/43/0/sceneId: {
  time: 1706734599724,
  value: undefined,
  nodeName: 'Telecommande',
  nodeLocation: ''
} with options { qos: 1, retain: false }
```

Here is the full Hass Device Json
```
{
  "type": "number",
  "object_id": "config_number_16",
  "discovery_payload": {
    "value_template": "{{ value_json.value }}",
    "command_topic": "zwave/27/112/0/16/set",
    "enabled_by_default": false,
    "entity_category": "config",
    "min": 0,
    "max": 5,
    "state_topic": "zwave/27/112/0/16",
    "availability": [
      {
        "payload_available": "true",
        "payload_not_available": "false",
        "topic": "zwave/27/status",
        "value_template": "{{'true' if value_json.value else 'false'}}"
      },
      {
        "topic": "zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/status",
        "value_template": "{{'online' if value_json.value else 'offline'}}"
      },
      {
        "payload_available": "true",
        "payload_not_available": "false",
        "topic": "zwave/driver/status"
      }
    ],
    "availability_mode": "all",
    "device": {
      "identifiers": [
        "zwavejs2mqtt_0xc7bf3f61_node27"
      ],
      "manufacturer": "Z-Wave.Me",
      "model": "Z-Wave Remote Control+ (ZME_RC2)",
      "name": "Telecommande",
      "sw_version": "1.1"
    },
    "name": "Telecommande_config_number_16",
    "unique_id": "zwavejs2mqtt_0xc7bf3f61_27-112-0-16"
  },
  "discoveryTopic": "number/Telecommande/config_number_16/config",
  "values": [
    "112-0-16"
  ],
  "persistent": false,
  "ignoreDiscovery": false,
  "id": "number_config_number_16"
}
```

Tell me if I have to send other informations.

Thank you for your help

### PR DESCRIPTION
**Pull request by ccutrer**
as a number

fixes #3570

Tested against openHAB and Home Assistant.

### CODE REVIEW

**Comment by apella12** [comment]
FYI I can't help on the coding and not overly familiar with these CCs, but will comment that scene activation and scene control CCs both launch scenes and probably should be setup the same.  The difference is that scene control is via the devices' Association groups and scene activation is via the lifeline to the controller.  From the specs.
```
2.2.85 Scene Controller Configuration Command Class, version 1
The Scene Controller Configuration Command Class is used to configure nodes launching scenes using
their association groups
2.2.85.2 Scene Controller Configuration Set Command
This command is used to configure settings for a given physical item on the device.
Bytes:
Command Class = COMMAND_CLASS_SCENE_CONTROLLER_CONF
Command = SCENE_CONTROLLER_CONF_SET
Group ID
Scene ID
Dimming Duration
------
2.2.83 Scene Activation Command Class, version 1
The Scene Activation Command Class used for launching scenes in a number of actuator nodes e.g.
another scene-controlling unit, a multilevel switch, a binary switch etc.
2.2.83.2 Scene Activation Set Command
This command is used to activate the setting associated to the scene ID.
Bytes:
Command Class = COMMAND_CLASS_SCENE_ACTIVATION
Command = SCENE_ACTIVATION_SET
Scene ID
Dimming Duration
------
```
On an unrelated note, the openHab Zwave binding supports either option.

File: api/lib/Gateway.ts [hunk 1]
```diff
@@ -1313,6 +1313,24 @@ export default class Gateway {
 						cfg = this._addRgbColorSwitch(node, valueId)
 					} else return
 					break
+				case CommandClasses['Scene Activation']:
+					cfg = utils.copy(hassCfg.scene_activation)
+					if (valueId.writeable) {
+						cfg.discovery_payload.command_topic = true
+					} else {
```
[the remark this answers is not in the document]
Comment (ccutrer) [reply api/lib/Gateway.ts:1325]: Good to know. I wasn't sure. For my devices, on the Central Scene CC (which I would normally assume to be read only), they have two values that are read-only, and one that is read-write, so I wanted to be defensive.

**Review by robertsLando — APPROVED** [review]
LGTM, I would like if possible if also @kpine can do a review

**Comment by ccutrer** [comment]
I don't see how fixing something that couldn't have worked is a breaking change. It's a bug fix.

**Comment by ccutrer** [comment]
The entire point of Scene Activation CC is to send a command to the device to trigger a scene. A sensor component is read only. You could read the last scene that was manually set on the device, but not actually send a scene command to it (at least when using Home Assistant or openHAB).

**Comment by ccutrer** [comment]
Ahhhh, now _that_ is a critical piece of information I was missing. My devices don't do that, and I didn't even know it was possible. My devices all send button presses to the controller with Central Scene commands, and when talking directly with each other use Basic Set or Multilevel Set commands, but don't communicate that to the controller when they do so. The only purpose for Scene Activation is to tell the switch to activate to a particular preset (and you could use a multicast to the same scene to several switches, which each have their own presets for a given scene).

So are you saying some devices will update the controller with their current Scene Activation value, and that value can be triggered locally (like say if scene 1 is dimming to 50%, when you set to 50% it will update the controller and say "I'm now at scene 1")? Or are you saying "pressing the top left button on a device with 6 buttons will send a Scene Activation value of 1 to the controller"? Or are you saying "I can configure my switch to directly control another switch, and when I use that button on the switch, on or both of those switches will notify the controller that this happened via the Scene Activation value"?

In that case, not only is the component type is a breaking change, but also the component ID is a breaking change. In order to maintain compatibility, it will still need to publish a Scene Activation CC under the ID the Central Scene component uses.

**Comment by ccutrer** [comment]
Closed in favor of a full revert: #3583

**Pull request closed without merging**